### DISCUSSION TITLE
Pinout Pimoroni Pirate Audio & RC522 not-usb Reader

### DISCUSSION THREAD

**Discussion by clemsonfire** [MiczFlor/RPi-Jukebox-RFID#1079] [Q&A] [answered]
Guten Abend!

In meiner Anfrage #1074 wurde mir großartig von Euch geholfen - nun plagt mich das nächste Problem.

Ich habe mir jetzt mal die Pinbelegungen des NICHT-USB RFID Readers und jener des Pimoroni Audio HATs angeschaut. https://tinyurl.com/y3ldqkf6 (Reader an Raspi Zero) und https://pinout.xyz/pinout/pirate_audio_speaker (Pirate Audio).

**Meine Hardware: RASPI Zero, PIRATE AUDIO 3W, RC522 Cardreader**

Dabei musste ich feststellen, dass 5 Raspi Zero Anschlüsse sowohl vom Reader wie auch vom Pirate HAT benötigt werden.

Auch wenn man auf das Display verzichten würde - wäre okay, ist aber doch ein nettes feature - hätte man noch m´immer mit einem PIN das Problem.

Könnt Ihr mir bitte sagen, wie man das lösen kann?

Und: Ich habe mir die "normale" Version installiert. Jetzt sehe ich, dass für die Pirate Installationsskripts "DEVELOP" empffohlen wird... kann man dieses Script dennoch benutzen?

Bitte um Antworten und Eure Tricks!

Im Voraus dankt Clemens

**Comment by 0x0lli** [comment]
hallo,

https://github.com/MiczFlor/RPi-Jukebox-RFID/wiki/Hardware-Pinout-Overview

ohne genaue Kenntnis von der Gesamtthematik mit dem pirateaudio zu haben, erscheint mir, Du seist mit dem neuftech RFID Reader besser bedient, solltest Du weiterhin keine Resonanz von Leuten, die das GPIO mapping auf Systemebene nachvollziehen können, bekommen. ich bin leider keiner davon.
mit dem neuftech kämst Du mit einem OTG adapter für microUSB -> USB-A hin.

Ich war gestern schon drauf und dran,  Deinen thread zu kapern, weil ich nicht mal den RFID-RC522 zum laufen bekommen habe OHNE hifiberry miniamp. der miniamp ging mal, dann ging er beim nächsten neustart wieder nicht. war ein Albtraum und wie sich später herausstellte, lag es an unzuverlässigen Jumperkabeln. darum heute ein extension board genommen und ohne bad cable 0,96" OLED und RFID-RC522 mit dem hifiberry miniamp zum laufen gebracht. jetzt sind beide 3,3V und 5V Leitungen belegt. 

nächste Baustelle: 
GPIO buttons ohne Löten? 
Kann man miniamp, rc-rfid522, OLED UND rotary encoder zusammen betreiben oder kommt man da langsam in Stromnot?

**Comment by OITDE** [comment]
Hi!
Auch wenn der Thread älter ist, stehe ich vor dem gleichen Problem. Gibt es dort eine Lösung? Ansonsten ist es wohl doch einfacher, den USB RFID Reader zu besorgen.

Viele Grüsse, Ingo

**Comment by Monikajosa** [comment]
Hey is there now a Working solution for Pirate Audio Hat and an RFID RC522 or any other RFID Reader whicht works with place - Capable?

**Comment by ChristianHerget** [comment, marked as answer]
Hello I also tried using the RC522 together with the PirateAudio display and they don't work together. The issue is probably that the PirateAudio uses GPIO 9 as DC signal to the display, but that's actually SPI0 MISO (Master In Slave Out). If you now start the display driver it will reconfigure this SPI input pin as a GPIO output for the display and the RC522 will stop working. It would be required to reconfigure GPIO 9 on the fly, but than there is the risk that the two threads (one for RC522 and one for the display) are trying to access SPI0 at the same time.

The solution for me was to switch to a PN532 and using the I2C interface instead. I know that the RC522 also has an I2C interface, but the driver in future3 only supports SPI.
